src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
+import { useState, FormEvent } from 'react';
+import users from './api/users';
+import initialTodos from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+type Todo = {
+  id: number;
+  title: string;
+  userId: number;
+  completed: boolean;
+  user?: (typeof users)[number];
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const trimmedTitle = title
+      .replace(/[^a-zA-Zа-яА-ЯґҐєЄіІїЇ0-9 ]/g, '')
+      .trim();
+
+    setTitle(trimmedTitle);
+
+    const isTitleValid = trimmedTitle !== '';
+    const isUserValid = userId !== '';
+
+    setTitleError(!isTitleValid);
+    setUserError(!isUserValid);
+
+    if (!isTitleValid || !isUserValid) {
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id), 0);
+
+    const selectedUser = users.find(user => user.id === +userId);
+
+    const newTodo: Todo = {
+      id: maxId + 1,
+      title: trimmedTitle,
+      userId: +userId,
+      completed: false,
+      user: selectedUser,
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUserId('');
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleSubmit} action="/api/todos" method="POST">
+        <label htmlFor="titleInput">Title:</label>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            id="titleInput"
+            data-cy="titleInput"
+            placeholder="Enter title"
+            value={title}
+            onChange={e => {
+              setTitle(
+                e.target.value.replace(/[^a-zA-Zа-яА-ЯґҐєЄіІїЇ0-9 ]/g, ''),
+              );
+              if (titleError) {
+                setTitleError(false);
+              }
+            }}
```

Comment:
```suggestion
            onChange={handleTitleChange}
```
Create a handler functions instead of passing anonim functions to `onChange`